bin/Macronify/MacroniRewriters.cpp

```diff
@@ -0,0 +1,281 @@
+#include "MacroniRewriters.hpp"
+
+namespace macroni {
+    bool has_results(mlir::Operation *op) {
+        return op->getNumResults() > 0;
+    }
+
+    bool is_get_user(macroni::MacroExpansion &exp) {
```

Comment:
The mlir "concrete" operations such as `macroni::MacroExpansion" or `mlir::hl::FuncOp` are intended to be copied around (they are really lightweight, just some wrapped pointer).